pkg/generate/code/set_resource.go

```diff
@@ -1226,53 +1226,61 @@ func SetResourceForStruct(
 	sourceShape := sourceShapeRef.Shape
 	targetShape := targetShapeRef.Shape
 
-	for memberIndex, memberName := range sourceShape.MemberNames() {
-		targetMemberShapeRef := targetShape.MemberRefs[memberName]
-		if targetMemberShapeRef == nil {
+	for _, targetMemberName := range targetShape.MemberNames() {
```

Comment:
This will be just fine for now (and is a perfect stepping stone for future commits, as you allude to in your PR summary). Long-term, I'd be keen to change this to loop over `r.Fields[targetMemberName].MemberFields`, though, which will contain all the additional custom fields as well.